discord/automod.py

```diff
@@ -0,0 +1,463 @@
+        exempt_channels: Sequence[:class:`abc.Snowflake`]
+            The new channels to exempt from the rule.
+        reason: :class:`str`
+            The reason for updating this rule. Shows up on the audit log.
+
+        Raises
+        -------
+        Forbidden
+            You do not have permission to edit this rule.
+        HTTPException
+            Editing the rule failed.
+
+        Returns
+        --------
+        :class:`AutoModRule`
+            The updated auto moderation rule.
+        """
+        transformed_actions = None
+        if actions:
+            transformed_actions = [action.to_dict() for action in actions]
+
+        data = await self._state.http.modify_auto_moderation_rule(
+            self.guild.id,
+            self.id,
+            name=name,
+            event_type=event_type,
+            actions=transformed_actions,
+            enabled=enabled,
+            exempt_roles=exempt_roles,
+            exempt_channels=exempt_channels,
+            reason=reason,
+        )
+
+        return AutoModRule(data=data, guild=self.guild, state=self._state)
+
+    async def delete(self, *, reason: str = MISSING) -> None:
+        """|coro|
+
+        Deletes the auto moderation rule.
+
+        Parameters
+        -----------
+        reason: :class:`str`
+            The reason for deleting this rule. Shows up on the audit log.
+
+        Raises
+        -------
+        Forbidden
+            You do not have permissions to delete the rule.
+        HTTPException
+            Deleting the rule failed.
+        """
+        await self._state.http.delete_auto_moderation_rule(self.guild.id, self.id, reason=reason)
+
+
+class AutoModAction:
+    """Represents an action that was taken as the result of a moderation rule.
+
+    Attributes
+    -----------
+    action: :class:`AutoModRuleAction`
+        The action that was taken.
+    message_id: Optional[:class:`int`]
+        The message ID that triggered the action.
+    rule_id: :class:`int`
+        The ID of the rule that was triggered.
+    rule_trigger_type: :class:`AutoModRuleTriggerType`
+        The trigger type of the rule that was triggered.
+    guild_id: :class:`int`
+        The ID of the guild where the rule was triggered.
+    user_id: :class:`int`
+        The ID of the user that triggered the rule.
+    channel_id: :class:`int`
+        The ID of the channel where the rule was triggered.
+    alert_system_message_id: Optional[:class:`int`]
+        The ID of the system message that was sent to the predefined alert channel.
+    content: :class:`str`
+        The content of the message that triggered the rule.
+    matched_keyword: Optional[:class:`str`]
+        The matched keyword from the triggering message.
+    matched_content: Optional[:class:`str`]
+        The matched content from the triggering message.
+
+
+    .. note::
+        The :attr:`content` requires the :attr:`Intents.message_content` or it will always return an empty string.
```

Comment:
This shouldn't be an admonition and it should be located in the paragraph in the fields that it applies to (which now includes `matched_content`).